power_grid_model_c/power_grid_model/include/power_grid_model/math_solver/y_bus.hpp

```diff
@@ -329,16 +339,90 @@ template <bool sym> class YBus {
         for (Idx entry = 0; entry != nnz(); ++entry) {
+            // start admittance accumulation with zero
+            ComplexTensor<sym> entry_admittance{0.0};
+            IdxVector entry_param_shunt;
+            IdxVector entry_param_branch;
+            //   loop over all entries of this position
+            for (Idx element = y_bus_entry_indptr[entry]; element != y_bus_entry_indptr[entry + 1]; ++element) {
+                auto param_idx = y_bus_element[element].idx;
+                if (y_bus_element[element].element_type == YBusElementType::shunt) {
+                    // shunt
+                    entry_admittance += shunt_param[param_idx];
+                    entry_param_shunt.push_back(param_idx);
+                } else {
+                    // branch
+                    entry_admittance +=
+                        branch_param[param_idx].value[static_cast<Idx>(y_bus_element[element].element_type)];
+                    entry_param_branch.push_back(param_idx);
+                }
+            }
+            // assign
+            admittance[entry] = entry_admittance;
+            map_admittance_param_branch.push_back(entry_param_branch);
+            map_admittance_param_shunt.push_back(entry_param_shunt);
+        }
+        // move to shared ownership
+        admittance_ = std::make_shared<ComplexTensorVector<sym> const>(std::move(admittance));
+    }
+
+    IdxVector increments_to_entries(auto const& math_model_param_incrmt) {
+        // construct affected entries
+        IdxVector affected_entries;
+
+        auto query_params_in_map = [&affected_entries](auto const& params_to_change, auto const& mapping) {
+            for (size_t i = 0; i < mapping.size(); ++i) {
+                if (std::ranges::any_of(mapping[i], [&](Idx val) {
+                        return std::ranges::find(params_to_change, val) != params_to_change.end();
+                    })) {
+                    affected_entries.push_back(static_cast<int>(i));
+                }
+            }
+        };
+
+        query_params_in_map(math_model_param_incrmt->branch_param_to_change, map_admittance_param_branch);
+        query_params_in_map(math_model_param_incrmt->shunt_param_to_change, map_admittance_param_shunt);
+        return affected_entries;
+    }
+
+    /**
+     * @brief Updates the admittance of the y_bus according to what's changed in math_model.
+     *
+     * @param math_model_param Shared pointer to the constant math_model parameters.
+     * @param math_model_param_incrmt Shared pointer to the constant mathematical model parameters .
+     */
+    void
+    update_admittance_increment(std::shared_ptr<MathModelParam<sym> const> const& math_model_param,
+                                std::shared_ptr<MathModelParamIncrement<sym> const> const& math_model_param_incrmt) {
+
+        // swap the old cached parameters
+        math_model_param_ = math_model_param;
+        math_model_param_incrmt_ = math_model_param_incrmt;
```

Comment:
with decrement gone, is there still a need to cache these?

Reply:
`math_model_param_incrmt_` not completely necessary indeed, but `math_model_param_ ` is just as not necessary, which was cached in the before version already for reason unknown to me.

Reply:
Resolved in latest commit.